src/test/java/edu/harvard/iq/dataverse/mydata/SolrQueryFormatterTest.java

```diff
@@ -127,12 +133,103 @@ private void msgt(String s){
+                { null,                                         "",          "dvObjectType", "sliceOfIds cannot be null",                       NullPointerException.class },
+                { null,                                         "paramName", null,           "sliceOfIds cannot be null",                       NullPointerException.class },
+                { null,                                         "paramName", "",             "sliceOfIds cannot be null",                       NullPointerException.class },
+                { null,                                         "paramName", "dvObjectType", "sliceOfIds cannot be null",                       NullPointerException.class },
+
+                { new ArrayList<Long>(),                        null,        null,           "paramName cannot be null",                        NullPointerException.class },
+                { new ArrayList<Long>(),                        null,        "",             "paramName cannot be null",                        NullPointerException.class },
+                { new ArrayList<Long>(),                        null,        "dvObjectType", "paramName cannot be null",                        NullPointerException.class },
+                { new ArrayList<Long>(),                        "",          null,           "sliceOfIds must have at least 1 value",           IllegalStateException.class },
+                { new ArrayList<Long>(),                        "",          "",             "sliceOfIds must have at least 1 value",           IllegalStateException.class },
+                { new ArrayList<Long>(),                        "",          "dvObjectType", "sliceOfIds must have at least 1 value",           IllegalStateException.class },
+                { new ArrayList<Long>(),                        "paramName", null,           "sliceOfIds must have at least 1 value",           IllegalStateException.class },
+                { new ArrayList<Long>(),                        "paramName", "",             "sliceOfIds must have at least 1 value",           IllegalStateException.class },
+                { new ArrayList<Long>(),                        "paramName", "dvObjectType", "sliceOfIds must have at least 1 value",           IllegalStateException.class },
+
+                { new ArrayList<Long>() {{ add(null); }},       null,        null,           "paramName cannot be null",                        NullPointerException.class },
+                { new ArrayList<Long>() {{ add(null); }},       null,        "",             "paramName cannot be null",                        NullPointerException.class },
+                { new ArrayList<Long>() {{ add(null); }},       null,        "dvObjectType", "paramName cannot be null",                        NullPointerException.class },
+                { new ArrayList<Long>() {{ add(null); }},       "",          null,           "(:())",                                           null },
```

Comment:
The expected query part seems not to make sense.
=> Check if there are only undefined ids (`null`) and throw an exception if there is no id defined as `Long`.